libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2592,6 +2579,25 @@ void IRGeneratorForStatements::appendBareCall(
 	appendCode() << templ.render();
 }
 
+void IRGeneratorForStatements::assignInternalFunctionIDIfNotCalledDirectly(
+	Expression const& _expression,
+	FunctionDefinition const& _referencedFunction
+)
+{
+	solAssert(
+		dynamic_cast<MemberAccess const*>(&_expression) ||
+		dynamic_cast<Identifier const*>(&_expression),
+		""
+	);
+	if (_expression.annotation().calledDirectly)
+		return;
+
+	define(IRVariable(_expression).part("functionIdentifier")) <<
+		to_string(m_context.internalFunctionID(_referencedFunction, false)) <<
```

Comment:
`define()` returns `std::ostream` and `internalFunctionID()` returns `uint64_t`. Is `to_string()` actually needed here?

Reply:
`ostream` can have a weird internal state that causes all sorts of formatting to take place, we certainly do not want that. I really hate that about the C++ streams...

Reply:
True. I hit that one in tests when I wanted to print bools as bools rather than 0 and 1. There's an option for that but it's persistent so I had to awkwardly check the flag and restore the original value afterwards.

Let's leave it as is then.